src/utils/datetime.ts

```diff
@@ -0,0 +1,251 @@
+import isNaN from 'lodash/isNaN';
+
+const MILLISECONDS_PER_SECOND = 1000;
+// 24 hours * 60 minutes * 60 seconds * 1000 milliseconds
+const MILLISECONDS_PER_DAY = 24 * 60 * 60 * MILLISECONDS_PER_SECOND;
+// 60 sec * 1000
+const MILLISECONDS_PER_MINUTE = 60 * MILLISECONDS_PER_SECOND;
+
+/**
+ * RegExp matcher for acceptable ISO 8601 date formats w/ timezone (see below)
+ * Capture groups structured as follows:
+ * 1) the date/time portion (2018-06-13T00:00:00.000)
+ * 2) the milliseconds (if matched)
+ * 3) the timezone portion (e.g., Z, +03, -0400, +05:00)
+ * 4) the Z format for timezone (if matched)
+ * 5) the short format for timezone (if matched)
+ * 6) the colon-less format for timezone (if matched)
+ * 7) the colon long format for timezone (if matched)
+ */
+const RE_ISO8601_DATE =
+    /^(\d{4}-\d{2}-\d{2}T\d{2}:\d{2}:\d{2}(\.\d{1,3})?)?((Z$)|(?:[+-](?:([0-2]\d$)|([0-2]\d(?:00|30)$)|([0-2]\d:(?:00|30)$))))$/;
```

Comment:
I assume this regex is copy/pasted from some previous working example?

Reply:
yea